marimo/_plugins/ui/_impl/refresh.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2023 Marimo. All rights reserved.
+from __future__ import annotations
+
+from typing import Final, Optional, Union
+
+from marimo._output.rich_help import mddoc
+from marimo._plugins.ui._core.ui_element import UIElement
+
+
+@mddoc
+class refresh(UIElement[int, int]):
+    """
+    A refresh button that will auto-refresh its descendants for a
+    given interval.
+    Each option value can either be a number (int or float) in seconds or a
+    human-readable string (e.g. "1s", "10s", "1m").
+
+    You can also combine multiple time units (e.g. "1m 30s").
+
+    **Example.**
+    ```python
+    refresh_button = mo.ui.refresh_button(
+        options=["1m", "5m 30s", "10m"],
+        default_interval="10m",
+    )
+    ```
+    **Attributes.**
+    - `options`: The options for the refresh interval, as a list of
+    human-readable strings or numbers (int or float) in seconds.
+    - `default_interval`: The default value of the refresh interval.
+    - `value`: The time in seconds since the refresh has been activated.
+
+    **Initialization Args.**
+    - `options`: The options for the refresh interval, as a list of
+    human-readable strings or numbers (int or float) in seconds.
+    If no options are provided and default_interval is provided,
+    the options will be generated automatically.
+    If no options are provided and default_interval is not provided,
+    the refresh button will not be displayed with a dropdown for auto-refresh.
+    - `default_interval`: The default value of the refresh interval.
+    """
+
+    name: Final[str] = "marimo-refresh"
+
+    def __init__(
+        self,
+        options: Optional[list[Union[int, float, str]]] = None,
+        default_interval: Optional[Union[int, float, str]] = None,
```

Comment:
For consistency with other elements, let's add a label argument? Could see a user doing something like:

`mo.ui.refresh(label="reload data")`